src/Raven.Server/Documents/Indexes/IndexStore.cs

```diff
@@ -356,6 +362,17 @@ private void HandleChangesForStaticIndexes(DatabaseRecord record, long index, Li
             }
         }
 
+        private bool ShouldSkipThisNode(DatabaseRecord record, string indexName)
+        {
+            if (record.StaggeredIndexes.TryGetValue(indexName, out var staggeredIndexesByNodeTag) == false)
+                return false; // Can this happen? Should we throw or ignore?
```

Comment:
Yes, this can happen if there is a race in the record. Shouldn't throw, just start processing it.